src/Point/Point.cs

```diff
@@ -81,15 +86,116 @@ public async Task SimulateTrailed()
+            _pointState.LatestFinalPointPosition = reportedPointPosition switch
+            {
+                ReportedPointPosition.PointIsInARightHandPositionDefinedEndPosition => reportedPointPosition,
+                ReportedPointPosition.PointIsInALeftHandPositionDefinedEndPosition => reportedPointPosition,
+                _ => _pointState.LatestFinalPointPosition,
+            };
+        }
+
+
+        /// <summary>
+        /// Sets the point into a degraded state. 
+        /// <c>ReportedPointPosition</c> <param name="message">the PointDegradedMessage.</param>
+        /// </summary>
+        /// <returns></returns>
+        public async Task SetDegraded(PointDegradedMessage message)
+        {
+            _pointState.PointPosition = ReportedPointPosition.PointIsInNoEndPosition;
+
+            if (_currentConnection != null)
+            {
+                ReportedDegradedPointPosition? reportedDegradedPointPosition = _hasNoNonCrucialPointMachines ?
+                    ReportedDegradedPointPosition.DegradedPointPositionIsNotApplicable :
+                    GetReportedDegradedPointPosition(_pointState.LatestFinalPointPosition);
+
+                if (reportedDegradedPointPosition != null)
+                {
+                    ReportedPointPosition? reportedPointPosition = GetReportedPointPositionDegraded(message.Position);
+                    if (reportedPointPosition != null)
+                    {
+                        UpdatePointState((ReportedPointPosition)reportedPointPosition, (ReportedDegradedPointPosition)reportedDegradedPointPosition);
+                        var occupancyStatus = new PointPointPositionMessage(_localId, _remoteId, _pointState.PointPosition, _pointState.DegradedPointPosition);
+                        await _currentConnection.RequestStream.WriteAsync(new SciPacket() { Message = ByteString.CopyFrom(occupancyStatus.ToByteArray()) });
+                    }
+                }
+            }
+
+            await UpdateConnectedWebClients();
+        }
+
+        /// <summary>
+        /// Sets the <c>ReportedPointPosition</c> to the latest final position (left/right).
+        /// </summary>
+        /// <returns></returns>
+        public async Task FinalizePosition(){
+            
+            if (_currentConnection != null)
+            {
+                if (_pointState.PointPosition != ReportedPointPosition.PointIsInARightHandPositionDefinedEndPosition &&
+                _pointState.PointPosition != ReportedPointPosition.PointIsInALeftHandPositionDefinedEndPosition) {
+                    ReportedDegradedPointPosition reportedDegradedPointPosition = _hasNoNonCrucialPointMachines ?
+                            ReportedDegradedPointPosition.DegradedPointPositionIsNotApplicable : ReportedDegradedPointPosition.PointIsNotInADegradedPosition;
+                    
+                    UpdatePointState(_pointState.LatestFinalPointPosition, reportedDegradedPointPosition);

```

Comment:
let's remove latestfinalpointposition and derive the value from the degraded position